Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -578,29 +590,30 @@ void ReflectometryReductionOne::exec() {
     auto detectorWS = inLam.get<0>();
     auto monitorWS = inLam.get<1>();
 
-    if (isMultiDetector) {
-      if (directBeam.is_initialized()) {
-        // Sum over the direct beam.
-        WorkspaceIndexList db = directBeam.get();
-        std::stringstream buffer;
-        buffer << db.front() << "-" << db.back();
-        MatrixWorkspace_sptr regionOfDirectBeamWS = this->toLamDetector(
-            buffer.str(), runWS, wavelengthInterval, wavelengthStep);
+    if (detectorEfficiencyCorrection.is_initialized()) {
+      MatrixWorkspace_sptr decWS = *detectorEfficiencyCorrection;
+      if (decWS->blocksize() > 1) {
+        // If using separate normalization constants for each spectra, rebin
+        // to the detector workspace
+        if (decWS->blocksize() < detectorWS->blocksize()) {
+          throw std::invalid_argument("The number of y-values in the detector "
```

Comment:
To summarize what we want:

- Case 1: Input workspace has M spectra (the number of bins is irrelevant) and flood workspace has M spectra with one bin in each spectrum. A script exemplifying what we want:

```
M=10

# Input workspace: 10 Spectra - 6 bins
x1 = [0,1,2,3,4,5,6] * M
y1 = [1,1,1,1,1,1,1] * M
ws1 = CreateWorkspace(x1, y1, NSpec=M)

# Flood workspace: 10 Spectra - 1 bin
x2 = [0] * M
y2 = [2] * M
ws2 = CreateWorkspace(x2, y2, NSpec=M)

ws3 = Divide(ws1, ws2)
```

- Case 2: Input workspace has M spectra (number of bins is irrelevant) and flood workspace has one spectrum with a single value. Example:

```
M=10

# Input workspace: 10 Spectra - 6 bins
x1 = [0,1,2,3,4,5,6] * M
y1 = [1,1,1,1,1,1,1] * M
ws1 = CreateWorkspace(x1, y1, NSpec=M)

# Flood workspace: 1 spectrum - 1 bin
x2 = [0]
y2 = [2]
ws2 = CreateWorkspace(x2, y2)

ws3 = Divide(ws1, ws2)
```

In both cases we can directly run `Divide` without the need to convert into the original form of 1-spectrum-many-values.